main.go

```diff
@@ -25,16 +29,31 @@ func main() {
 
 	swaggerFile := flag.String("file", "swagger.yml", "the spec file to use")
 	goPackageName := flag.String("go-package", "", "package of the generated go code")
+	outputPath := flag.String("output-path", "", "relative output path of the generated go code")
 	jsModulePath := flag.String("js-path", "", "path to put the js client")
 	versionFlag := flag.Bool("version", false, "print the wag version and exit")
 	flag.Parse()
 	if *versionFlag {
 		fmt.Println(version)
 		os.Exit(0)
 	}
-	if *goPackageName == "" {
-		log.Fatal("go-package is required")
+
+	var goPackagePath string
+	// check if the repo uses modules
+	if modFile, err := os.Open("go.mod"); err != nil {
+		if _, ok := err.(*os.PathError); !ok {
+			log.Fatalf("Error checking if the repo contains a go.mod file: %s", err.Error())
+		}
+		if *goPackageName == "" {
+			log.Fatal("go-package is required")
+		}
+		goPackagePath = *goPackageName
```

Comment:
Only an os.PathError should lead to that path (line 47)

- if the error is not nil and not a os.PathError, then we error out because is is an unknown error
- if the error is not nil and a os.pathError, that means there is no go.mod file and execution continues

Reply:
oops, I'm stupid and didn't see the `!`